### PR TITLE
Find and replace: toolbar button is now using DropdownButtonView

### LINKED ISSUE #10007: The find and replace toolbar button is using a split button instead of dropdown button
## 📝 Provide detailed reproduction steps (if any)

1.  http://fake.ckeditor.com:8081/build/docs/ckeditor5/latest/features/find-and-replace.html

### ✔️ Expected result

The find and replace button is using a `DropdownButtonView` same as `insertTable` button.

### ❌ Actual result

It's using a `SplitButtonView` and action for both sides is the same.

### ❓ Possible solution

_If you have ideas, you can list them here. Otherwise, you can delete this section._

## 📃 Other details

*   Browser: …
*   OS: …
*   First affected CKEditor version: …
*   Installed CKEditor plugins: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by mateuszzagorski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (find-and-replace): Find and replace toolbar button now uses `DropdownButtonView`. Closes #10007.

### CODE REVIEW

File: packages/ckeditor5-find-and-replace/tests/findandreplace.js [hunk 1]
```diff
@@ -116,16 +116,6 @@ describe( 'FindAndReplace', () => {
...
 			it( 'has a proper initial state', () => {
-				// "Initial state" mock.
-				editor.setData( LONG_TEXT );
-
-				const itemView = Array.from( editor.ui.view.toolbar.items )
-					.filter( item =>
-						item.buttonView && item.buttonView.tooltip == 'Find and replace'
-					)[ 0 ];
-
-				itemView.buttonView.arrowView.fire( 'execute' );
```
Comment (mlewand) [inline packages/ckeditor5-find-and-replace/tests/findandreplace.js:127]: Why was this preparation logic removed?

This test was meant to test initial state after the dropdown is shown. With this change we check just state after formView was constructed but not really shown.

Comment (mlewand) [reply packages/ckeditor5-find-and-replace/tests/findandreplace.js:127]: This question is valid for other instances like that.

**Review by mlewand — COMMENTED** [review]
The fix remains valid however I have questions regarding changes in unit tests.

File: packages/ckeditor5-find-and-replace/tests/findandreplace.js (hunk 1 above)
Comment (mateuszzagorski) [reply packages/ckeditor5-find-and-replace/tests/findandreplace.js:127]: You are right, that logic should've stayed there. I've reverted the changes made before.

**Review by mlewand — APPROVED** [review]

**Pull request merged**